src/isar/state_machine/state_machine.py

```diff
@@ -257,11 +259,20 @@ def _mission_finished(self) -> None:
         partially_fail_statuses = fail_statuses + [TaskStatus.PartiallySuccessful]
 
         if all(task.status in fail_statuses for task in self.current_mission.tasks):
+            self.current_mission.user_message = UserMessage(
+                full_text=f"The mission failed because all tasks in the mission failed",
+                short_text="all tasks in the mission failed",
+            )
             self.current_mission.status = MissionStatus.Failed
         elif any(
             task.status in partially_fail_statuses
             for task in self.current_mission.tasks
         ):
+            self.current_mission.user_message = UserMessage(
+                full_text="The mission was partially successful because one or more "
+                "tasks failed",
```

Comment:
Should we add the reasons from the failing tasks here (as done for steps in the task messages)?

Reply:
See above

Reply:
If we do the above, I think we should remove this description altogether, it's just describing the status without providing any information